src/sinks/new_relic_logs.rs

```diff
@@ -115,9 +119,8 @@ impl NewRelicLogsConfig {
 
         let batch = self.batch.use_size_as_bytes()?;
         let batch = BatchConfig {
-            // The max request size is 10MiB, so in order to be comfortably
-            // within this we batch up to 5MiB.
-            max_bytes: Some(batch.max_bytes.unwrap_or(bytesize::mib(5u64) as usize)),
+            // New Relic Logs API accepts payloads only up to 1MB (10⁶ bytes)
+            max_bytes: Some(cmp::min(batch.max_bytes.unwrap_or(MAX_PAYLOAD_SIZE), MAX_PAYLOAD_SIZE)),
```

Comment:
I think I'd prefer to raise an error if they pass a payload size that is too big rather than transparently cap it. I think that'll result in more easily discoverable behavior. This could be a new `BuildError`.

Reply:
I wish I knew more about what drives the payload size. 
Other New Relic plugins implement payload splitting, in case it's too big. 
Do you think we should do the same here?

Reply:
Do you have an example of a plugin that does that? That would increase my confidence regarding what it considers to be part of the payload in its calculation.

For payload splitting, the `batch` configuration that you are modifying here should handle that for us. It will only batch together events whose lengths sum to less than X bytes (or `n` events if `max_events` is set too).

Reply:
Noting that the batching seems to be before compression. 1 MiB of events is likely to compress down to a much smaller payload. I opened https://github.com/timberio/vector/issues/4716 to think about how to improve this behavior.

Reply:
As for payload splitting, these plugins come into mind.
There's [the fluent-bit plugin, written in go](https://github.com/newrelic/newrelic-fluent-bit-output/blob/master/record/record.go#L89-L118). And [fluentd plugin in Ruby has a similar implementation](https://github.com/newrelic/newrelic-fluentd-output/blob/master/lib/fluent/plugin/out_newrelic.rb#L127-L176).

One thing to keep in mind, is that this code is best effort, and may not be perfect. But I hope it helps.

Reply:
Gotcha, thanks! Looking at the Go implementation, the cap seems to be with respect to the payload size rather than including any other parts of the request. :+1: